M2/autogen.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+echo "-- Generating configure script"
+autoreconf --verbose --force --install
+
+if test ! -f Macaulay2/editors/emacs/M2.el
+then
+    if test -d ../.git
+    then
+	echo "-- Updating M2-emacs submodule"
+	git submodule update --init Macaulay2/editors/emacs
+    else
+	echo "-- Warning: not in git repository"
+	echo "-- Downloading and extracting M2-emacs tarball"
```

Comment:
Is this necessary? e.g. what if they didn't want the emacs mode? I think a warning should suffice, no?